.github/workflows/_deploy_kafka_test.yml

```diff
@@ -0,0 +1,134 @@
+name: Deploy Kafka Test
+
+on:
+  workflow_call:
+    inputs:
+      cluster-name:
+        required: true
+        description: AKS Cluster Name With Randomly Generated Suffix
+        type: string
+      sidecar-registry:
+        required: true
+        default: "mcr.microsoft.com"
+        description: "The registry to get the image from"
+        type: string
+      key-release-image: 
+        required: true 
+        description: "The image of the SKR sidecar to use"
+        type: string
+      consumer-image: 
+        description: "Consumer Image"
+        required: true
+        type: string
+      producer-image: 
+        description: "Producer Image"
+        required: true
+        type: string
+  workflow_dispatch:
+    inputs:
+      cluster-name:
+        required: true
+        description: AKS Cluster Name With Randomly Generated Suffix
+        type: string
+      sidecar_registry:
+        required: true
+        default: "mcr.microsoft.com"
+        description: "The registry to get the image from"
+        type: string
+      key-release-image: 
+        required: true 
+        description: "The image of the SKR sidecar to use"
+        type: string
+      consumer-image: 
+        description: "Consumer Image"
+        required: true
+        type: string
+      producer-image: 
+        description: "Producer Image"
+        required: true
+        type: string
+jobs:
+  deploy-kafka:
+    name: Deploy Kafka Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          path: main
+
+      - name: Log into Azure
+        run: |
+          az login --service-principal \
+            --username ${{ secrets.AZURE_SERVICE_PRINCIPAL_APP_ID }} \
+            --password ${{ secrets.AZURE_SERVICE_PRINCIPAL_PASSWORD }} \
+            --tenant ${{ secrets.AZURE_SERVICE_PRINCIPAL_TENANT }}
+            
+      - name: Install Dependencies 
+        id: install-dependencies 
+        run: |
+            curl -L https://github.com/a8m/envsubst/releases/download/v1.2.0/envsubst-`uname -s`-`uname -m` -o envsubst
```

Comment:
@ksayid recommends adding --fail-with-body to all curl commands so that we can get an error output.

Reply:
this is done